### ISSUE TITLE
`no-commonjs` does not appear to work for TypeScript

### ISSUE THREAD

**Issue by hbergren** [benmosher/eslint-plugin-import#1709]
I have the following rule enabled:
```js
    // disallow require()
    // https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-commonjs.md
    'import/no-commonjs': 'error',
```

However, in my TypeScript source file, the following line that I expected to error does not:
```ts
import Boom = require('boom')
```

Should it?

**Comment by ljharb** [comment]
Yes, but what I'd suggest is enabling esModuleInterop and synthetic imports (so TS's module system isn't broken) and then use `import Boom from 'boom'` :-)

**Comment by hbergren** [comment]
This rule seems to do the job for TypeScript in `@typescript-eslint`:

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-require-imports.md

**Comment by ljharb** [comment]
Sounds like that's great inspiration for a fix here, then!
